### PR TITLE
Introducing Kubemodel blog post

### PR DESCRIPTION
**Pull request by spa-raj**
* Blog post introducing Kubemodel - Opencost Data Model 2.0 which I worked on as part of my LFX Fall 2025 Mentorship.

### CODE REVIEW

File: blog/2025-12-03-introducing-kubemodel/index.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+A robust data model needs comprehensive testing. The KubeModel work includes extensive integration tests validating:
+
+- **UID format compliance** against RFC 4122 UUID standards
+- **Cross-window consistency** for stable infrastructure resources
+- **Ephemeral resource handling** - recognizing that pods may be recreated between time windows
+- **Metric emission verification** across 31 metrics and 9 resource types
+
+These tests run across multiple time windows (1h, 6h, 24h) to ensure consistency and catch edge cases in real-world scenarios.
+
+## The Road Ahead
```
Comment (ameijer) [inline blog/2025-12-03-introducing-kubemodel/index.md]: this section feel very AI-ish. Nothing in here is regarding what you think the next steps are, just what has already been accomplished, right?

**Review by ameijer — COMMENTED** [review]
general feedback - this is a tech blog post, I suggest you embed code snippets of a few selected data structures and walk the users through it. EMbed in places like the o(1) discussions etc

File: blog/2025-12-03-introducing-kubemodel/index.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- **Volume & PVC** - Persistent storage resources
+- **ResourceQuota** - Namespace resource limits and requests
+
+### Flat Architecture for Performance
+
+Unlike nested hierarchies, KubeModel maintains a **flat map-based structure** enabling O(1) lookups while preserving resource relationships through UIDs and references. This design choice prioritizes:
+
+- **Fast access**: Constant-time lookups regardless of cluster size
+- **Clean relationships**: Resources reference each other via stable identifiers
+- **Efficient serialization**: The flat structure translates well to binary formats
```
Comment (Sean-Holcomb) [inline blog/2025-12-03-introducing-kubemodel/index.md]: we could also mention that it is use case agnostic

File: blog/2025-12-03-introducing-kubemodel/index.md (hunk 1 above)
Comment (spa-raj) [reply blog/2025-12-03-introducing-kubemodel/index.md]: yes, I will remove the integration tests section.
